catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/dialog/LegoSensorPortConfigDialogTest.java

```diff
@@ -0,0 +1,127 @@
+import android.support.test.espresso.Espresso;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.bricks.ChangeSizeByNBrick;
+import org.catrobat.catroid.devices.mindstorms.nxt.sensors.NXTSensor;
+import org.catrobat.catroid.ui.SpriteActivity;
+import org.catrobat.catroid.ui.settingsfragments.SettingsFragment;
+import org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils;
+import org.catrobat.catroid.uiespresso.formulaeditor.utils.FormulaEditorWrapper;
+import org.catrobat.catroid.uiespresso.testsuites.Cat;
+import org.catrobat.catroid.uiespresso.testsuites.Level;
+import org.catrobat.catroid.uiespresso.util.UiTestUtils;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.catrobat.catroid.uiespresso.formulaeditor.utils.FormulaEditorWrapper.onFormulaEditor;
+import static org.catrobat.catroid.uiespresso.ui.dialog.utils.LegoSensorPortConfigDialogWrapper.onLegoSensorPortConfigDialog;
+import static org.catrobat.catroid.uiespresso.ui.fragment.rvutils.RecyclerViewInteractionWrapper.onRecyclerView;
+
+@Category({Cat.AppUi.class, Level.Smoke.class})
+public class LegoSensorPortConfigDialogTest {
+
+	@Rule
+	public BaseActivityInstrumentationRule<SpriteActivity> baseActivityTestRule = new
+					BaseActivityInstrumentationRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION, SpriteActivity.FRAGMENT_SCRIPTS);
+
+	private NXTSensor.Sensor[] sensorMapping;
+	private boolean legoNXTBrickSetting;
+
+	@Before
+	public void setUp() throws Exception {
+		Script script = BrickTestUtils.createProjectAndGetStartScript("LegoSensorPortConfigDialogTest");
+		script.addBrick(new ChangeSizeByNBrick(0));
+
+		legoNXTBrickSetting = getNXTBrickSetting();
+		setNXTBrickSetting(true);
+
+		sensorMapping = SettingsFragment.getLegoNXTSensorMapping(InstrumentationRegistry.getTargetContext());
+
+		SettingsFragment.setLegoMindstormsNXTSensorMapping(InstrumentationRegistry.getTargetContext(),
+						new NXTSensor.Sensor[] {NXTSensor.Sensor.NO_SENSOR, NXTSensor.Sensor.NO_SENSOR,
+								NXTSensor.Sensor.NO_SENSOR, NXTSensor.Sensor.NO_SENSOR});
+
+		baseActivityTestRule.launchActivity();
+
+		onBrickAtPosition(1).onChildView(withId(R.id.brick_change_size_by_edit_text)).perform(click());
+
+		openDialog();
+	}
+
+	@Test
+	public void pressOKTest() {
+		onLegoSensorPortConfigDialog().performClickOnOK();
+		onLegoSensorPortConfigDialog();
```

Comment:
please make checks explicitely.
onView(withId(R.id.dfghjkl)) will always work in espresso because it doesnt do anything.

so it would be nicer if you would add an explicite statement that youre actually checking something here.
onLegoSensorPortConfigDialog()
     .checkDialogVisible() 
or something like that